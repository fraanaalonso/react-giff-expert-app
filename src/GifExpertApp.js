import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = () =>{

    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState(['One Punch']);
/*
    const handleAdd = () => {
        setCategories( [...categories, 'HunterXHunter'] ); //keep the previous categories and I add another one
        //setCategories(cats => [ ...cats, 'HunterXHunter']); we receive a callback
    }
*/ 
    return (

        <>
            <h2>GiffExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr></hr>

            <ol>
                { 
                    categories.map( (category, i) =>  
                       <GifGrid 
                            key={ category }
                            category={ category } 

                       
                       />
                    )
                }
            </ol>
        </>
    )
}
