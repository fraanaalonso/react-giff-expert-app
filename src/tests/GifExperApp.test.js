import { shallow } from 'enzyme'
import React from 'react'
import { GifExpertApp } from '../GifExpertApp'


describe('Testing GifExpertApp', ()=> {
    test('Must test matching Snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Must show a list of categories ', () => {
        const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })
    
    
})