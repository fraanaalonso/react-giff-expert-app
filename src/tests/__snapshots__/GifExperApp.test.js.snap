// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp Must show a list of categories  1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Testing GifExpertApp Must test matching Snapshot 1`] = `
<Fragment>
  <h2>
    GiffExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
